src/test/scala/uk/gov/nationalarchives/tdr/api/service/ValidateFileMetadataServiceSpec.scala

```diff
@@ -424,17 +434,44 @@ class ValidateFileMetadataServiceSpec extends AnyFlatSpec with MockitoSugar with
     val mockFileStatusRepository: FileStatusRepository = mock[FileStatusRepository]
     val fixedUUIDSource = new FixedUUIDSource()
     fixedUUIDSource.reset
-    val service = new ValidateFileMetadataService(mockCustomMetadataService, mockFileMetadataRepository, mockFileStatusRepository, fixedTimeSource, fixedUUIDSource)
+    val service = new ValidateFileMetadataService(mockCustomMetadataService, mockFileMetadataRepository, mockFileStatusRepository)
     val mockFields = mockCustomMetadataFields()
 
+    def createExpectedFilestatusRow(fileId: UUID, statusType: String, statusValue: String): Seq[FilestatusRow] = {
+      val mappedMetadataTypeValue = (statusValue, statusType) match {
+        case ("Completed", _)                      => "Completed"
+        case ("Incomplete", "ClosureMetadata")     => "Incomplete"
+        case ("Incomplete", "DescriptiveMetadata") => "NotEntered"
+        case ("Incomplete", _)                     => "Incomplete"
+        case ("NotEntered", _)                     => "NotEntered"
+      }
+      Seq(FilestatusRow(null, fileId, statusType, mappedMetadataTypeValue, null))
+    }
+
     def stubMockResponses(metadataRows: List[FilemetadataRow] = List()): Unit = {
-      when(mockFileStatusRepository.addFileStatuses(any[List[FilestatusRow]])).thenReturn(Future.successful(Nil))
+      def generateExpectedRows(input: List[AddFileStatusInput]): Future[Seq[FilestatusRow]] = {
+        val expectedRows = input.flatMap { addStatusInput =>
+          createExpectedFilestatusRow(addStatusInput.fileId, addStatusInput.statusType, addStatusInput.statusValue)
+        }
+        Future.successful(expectedRows)
+      }
+
+      when(mockFileStatusRepository.addFileStatuses(any[List[AddFileStatusInput]])).thenAnswer { invocation: InvocationOnMock =>
+        val input: List[AddFileStatusInput] = invocation.getArgument(0)
+        generateExpectedRows(input)
+      }
       when(mockFileStatusRepository.deleteFileStatus(any[Set[UUID]], any[Set[String]])).thenReturn(Future.successful(1))
       when(mockCustomMetadataService.getCustomMetadata).thenReturn(Future(mockFields))
       when(mockFileMetadataRepository.getFileMetadata(Some(any[UUID]), any[Option[Set[UUID]]], any[Option[Set[String]]])).thenReturn(Future(metadataRows))
     }
   }
 
+  private def convertFilestatusRowToAddFileStatusInput(filestatusRows: List[FilestatusRow]): List[AddFileStatusInput] = {
```

Comment:
"file status" should be camel case eg: `convertFileStatusRowToAddFileStatusInput`. 

Note: the object "File status" from the Tables is not because this is auto generated by Slick and we have no control over its naming conventions, ie `... : List[FilestatusRow]]): ...`